app/models/host.rb

```diff
@@ -194,6 +194,17 @@ def self.failover
     where(:failover => true)
   end
 
+  def self.all_unique_downcased_domains(includes = nil)
+    select_rows = select("lower(regexp_replace(split_part(hostname, ',', 1), '[^\.]*\.', '')) as domain")
+                    .select(:id).to_sql
+                    .sub!("SELECT", "DISTINCT ON (domain)")
+                    .sub!(/ FROM.*$/, '')
```

Comment:
I noticed the other day while browsing the `arel` codebase (like you do), there is a `distinct_on` method I might be able to leverage here.  Will look into doing that at some point.